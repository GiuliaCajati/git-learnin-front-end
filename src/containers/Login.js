import React, { useState } from 'react';
import Link from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import { login } from "../actions"
import { ConsoleWriter } from 'istanbul-lib-report';
import { useSelector, useDispatch } from 'react-redux'

const useStyles = makeStyles((theme) => ({
    form: {
        width: '100%', // Fix IE 11 issue.
        marginTop: "1%"
    }
    }))

    
    
    const Login = (props) => {
        const classes = useStyles();
        const [value, setvalue] = useState("")
        
        const dispatch = useDispatch();
    
        const callback = (e) =>{
            e.preventDefault();
            const user = {
                username: e.target.querySelector("#username").value,
                password: e.target.querySelector("#password").value
            }
            dispatch(login(user))
        }

    return (

        <div style = {{textAlign: 'center'}} data-testid="Login">
            <form 
                onSubmit = {callback} 
                style = {{
                    width: '100%', // Fix IE 11 issue.
                    marginTop: "5%",
                    textAlign: "center"
                }}
            >
                
                <TextField
                    style = {{marginRight: "1%"}}
                    id = "username"
                    label = 'username'
                    onChange={(e) => setvalue(e.target.value)}
                    inputProps={{
                        'data-testid': "usernameField" 
                    }}
                />
                <TextField
                    style = {{
                        marginRight: "1%",
                        marginLeft: "1%"}}
                    id = "password"
                    label = 'password'
                    inputProps={{
                        'data-testid': "passwordField" 
                    }}
                />

                <Button
                    type = "submit"  
                    style = {{ marginTop: ".7%"}}
                    variant = "contained"

                    data-testid= "loginSubmit"

                    >
                        Login
                </Button>


            </form>
            {/* Make sure the sign up link is going to /signup */}
            <Link href="signup"  variant="body2" data-testid = "toSignup" ><b>
                {"Need to Git-Learnin? Sign Up"}
            </b>        
            </Link>
        </div>
    );
}

export default Login;