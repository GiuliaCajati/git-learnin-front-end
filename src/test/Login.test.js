import React from 'react'
import ReactDOM from 'react-dom'
import {render, screen, cleanup} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Login from '../containers/Login'

beforeEach(() => {
    render(<Login />)
});

afterEach(() => {
    cleanup()
});

test('Checking that login component renders properly', () => {
    const div = document.createElement('div')
    const component = ReactDOM.render(<Login />, div)
})

// test('Checking that login component matches snapshot', () => {
//     const div = document.createElement('div')
//     const component = render(<Login />, div)
//     expect(component.container).toMatchSnapshot()
// })

test('Component has a form', () => {
    const login = screen.getByTestId('Login')
    expect(login).toContainHTML('form')
})

test('Submit button works', () => {
    const button = screen.getByTestId('loginSubmit')
    expect(button).toHaveAttribute('type', 'submit')
})

test('Fetch returns a user', async () => {
    // expect.assertions(1);
    // const data = await functions.fetchUser()
    // expect(data.name).toEqual('Leanne Graham')
})

test('Test for state change', () => {

})
