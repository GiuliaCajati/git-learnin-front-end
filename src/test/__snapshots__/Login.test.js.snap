// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking that login component matches snapshot 1`] = `
<div>
  <div
    data-testid="Login"
    style="text-align: center;"
  >
    <form
      style="width: 100%; margin-top: 5%; text-align: center;"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        style="margin-right: 1%;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="username"
          id="username-label"
        >
          username
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            data-testid="usernameField"
            id="username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
        style="margin-right: 1%; margin-left: 1%;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          password
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            data-testid="passwordField"
            id="password"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        data-testid="loginSubmit"
        style="margin-top: .7%;"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
      data-testid="toSignup"
      href="signup"
    >
      <b>
        Need to Git-Learnin? Sign Up
      </b>
    </a>
  </div>
</div>
`;
